src/include/kdbtypes.h

```diff
@@ -85,7 +89,7 @@ typedef unsigned char             kdb_octet_t;
 typedef signed short              kdb_short_t;
 typedef unsigned short            kdb_unsigned_short_t;
 
-#if SIZEOF_LONG == 4
+#if (LONG_MAX == +2147483647)
```

Comment:
SIZEOF_LONG is set by cmake and should be fixed that it works on apple, too

    src/include/CMakeLists.txt:check_type_size(long            SIZEOF_LONG)